docs/development_best_practices.md

```diff
@@ -0,0 +1,130 @@
+# Development Best Practices
+
+This document describes the best practices to follow when developing operators or other application components such as partitoners, stream codecs etc.
+
+## Operators
+
+These are general guidelines for all operators and the next sections talk about special considerations for input and output operators.
```

Comment:
"... operators and the next ..." reads better as "... operators; the following ...